tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/Emitter.scala

```diff
@@ -345,11 +345,27 @@ final class Emitter private (semantics: Semantics, outputMode: OutputMode,
 
     // Class definition
     if (linkedClass.hasInstances && kind.isAnyNonNativeClass) {
-      val ctor = classTreeCache.constructor.getOrElseUpdate(
-          classEmitter.genConstructor(linkedClass)(classCache))
+      val (linkedInlineableInit, linkedMemberMethods) =
+        classEmitter.extractInlineableInit(linkedClass)(classCache)
+
+      // JS constructor
+      val ctor = {
+        /* The constructor depends both on the class version, and the version
+         * of the inlineable init. When there is no inlineable init, we use ""
+         * as its version.
+         */
+        val ctorCache = classCache.getConstructorCache()
+        val optInlineableInitVersion =
+          linkedInlineableInit.fold[Option[String]](Some(""))(_.version)
```

Comment:
Yeah but ... that's not really something we do, having `""` as version. We might even have some implicit guarantee that a version is never `Some("")`.

If you prefer, it's still possible to be more robust with something like:
```scala
ctorVersion = linkedInlineableInit.fold[Option[String]] {
  linkedClass.version.map("1-" + _)
} { inlineableInitVersion =>
  mergeVersions(linkedClass.version, inlineableInitVersion).map("2-" + _)
}
```